src/components/ui/NumericText/tests/index.test.js

```diff
@@ -0,0 +1,142 @@
+          });
+        });
+      });
+      describe('when value is less than default 6 decimal places', () => {
+        [
+          ['0.0000001', '1.000e-7'],
+          ['0.00000001', '1.000e-8'],
+          ['0.000000123456', '1.235e-7'],
+        ].forEach(([val, expected]) => {
+          it(`should render the exponential value ${val}`, () => {
+            wrapper = shallow(<NumericText
+              {...props}
+              value={val}
+            />
+            );
+            expect(getInnerText(wrapper.find(SelectableText).html())).toContain(expected);
+          });
+        });
+      });
+    });
+    describe('with maxDecimalPlaces param', () => {
+      describe('when maxDecimalPlaces param is 3', () => {
+        [
+          ['10000', '10000'],
+          ['0.001', '0.001'],
+          ['0.0001', '1.000e-4'],
+          ['0.000000123456', '1.235e-7'],
+        ].forEach(([val, expected]) => {
+          it(`should render ${expected} for value ${val}`, () => {
+            wrapper = shallow(<NumericText
+              {...props}
+              value={val}
+              maxDecimalPlaces={3}
+            />
+            );
+            expect(getInnerText(wrapper.find(SelectableText).html())).toContain(expected);
+          });
+        });
+      });
+      describe('when maxDecimalPlaces param is 20', () => {
+        [
+          ['10000', '10000'],
+          ['0.001', '0.001'],
+          ['0.0001', '0.0001'],
+          ['0.000000123456', '0.000000123456'],
+          ['0.00000012345678910111', '0.00000012345678910111'],
+          ['0.0000000000000000000123456', '0.0000000000000000000123456'],
+          ['0.00000000000000000000123456', '1.235e-21'],
+        ].forEach(([val, expected]) => {
+          it(`should render ${expected} for value ${val}`, () => {
+            wrapper = shallow(<NumericText
+              {...props}
+              value={val}
+              maxDecimalPlaces={20}
+            />
+            );
+            expect(getInnerText(wrapper.find(SelectableText).html())).toContain(expected);
+          });
+        });
+      });
+    });
+    describe('when value is 0', () => {
+      it('should render 0', () => {
+        wrapper = shallow(<NumericText
+          {...props}
+          value={'0'}
+        />
+        );
+        expect(getInnerText(wrapper.find(SelectableText).html())).toEqual('0');
+      });
+    });
+  });
+  describe('currency type', () => {
+    describe('with maxDecimalPlaces param', () => {
+      it('should pass correct params to the #formatNumber()', () => {
+        wrapper = shallow(<NumericText
+          {...props}
+          type={'currency'}
+          value={'0.001'}
+          maxDecimalPlaces={3}
+        />
+        );
+        expect(getInnerText(wrapper.find(SelectableText).html())).toEqual('0.001 {&quot;style&quot;:&quot;currency&quot;,&quot;currency&quot;:&quot;USD&quot;,&quot;maximumFractionDigits&quot;:3}');
```

Comment:
Is it necessary, or significant, to verify this part of the innerText?

Reply:
Good spot. Yes, it needs to verify the stringified parameters passed to the `formatNumber` intl function. Since `formatNumber` is a mocked dependency, the simplest way I can find to verify the parameters to it is to verify the stringified parameter.